pkg/hub/localizer/localizer.go

```diff
@@ -331,12 +334,26 @@ func (l *Localizer) getOverrides(namespace string, feed appsapi.Feed) ([]appsapi
 		uid = manifests[0].UID
 	}
 
-	globs, err := l.globLister.List(labels.SelectorFromSet(labels.Set{
+	feedGlobs, err := l.globLister.List(labels.SelectorFromSet(labels.Set{
 		string(uid): feed.Kind,
 	}))
 	if err != nil {
 		return nil, err
 	}
+
+	globs := make([]*appsapi.Globalization, 0)
+	for _, glob := range feedGlobs {
+		clusters, err := l.mclsCache.ListByNamespace(namespace, glob.Spec.ClusterAffinity)
+		if err != nil {
+			return nil, err
+		}
+		if len(clusters) == 0 {
```

Comment:
This is not correct. What if the `clusterAffinity` does match a group of clusters, but none of them are in the namespace that we're processing.